spec/Section 4 -- Introspection.md

```diff
@@ -143,7 +143,7 @@ type __Type {
   enumValues(includeDeprecated: Boolean = false): [__EnumValue!]
 
   # should be non-null for INPUT_OBJECT only, must be null for the others
-  inputFields: [__InputValue!]
+  inputFields(includeDeprecated: Boolean = false): [__InputValue!]
```

Comment:
I'm not sure if you're looking for feedback on this or #525 but won't defaulting to excluding deprecated input fields/field arguments result in problematic introspection results for older clients? For example, if I deprecate a field argument in my API, then the field argument will disappear in older versions of GraphiQL hitting my API which is not what I want and is inconsistent with how GraphiQL handles deprecated fields/enum values. 

Possible alternatives I can think of:

1. Default `includeDeprecated` to `true` for `__Type.inputFields` and `__Field.args` - not ideal since it's inconsistent with the default behavior of `__Type.fields` and `__Type.enumValues`.
1. Default `includeDeprecated` to `true` for `__Type.inputFields`, `__Field.args`, `__Type.fields` and `__Type.enumValues` which is a breaking change but won't actually break clients that explicitly specify `includeDeprecated`. Introspection queries generated by graphql-js explicitly set `includeDeprecated` it to true (see [here](https://github.com/graphql/graphql-js/blob/master/src/utilities/getIntrospectionQuery.js#L70)) but I'm not sure about other clients.
1. Don't support server-side filtering to include/exclude deprecated input fields/arguments and use this new functionality to deprecate the `includeDeprecated` argument for `__Type.inputFields` and `__Field.args`.
1. Introduce a new argument (e.g. `excludeDeprecated`) to  `__Type.inputFields`, `__Field.args`, `__Type.fields` and `__Type.enumValues` with defaulting that doesn't filter out deprecated members and deprecate the old `includeDeprecated` argument. Queries that specify both the old and new arguments would results in runtime errors.

Reply:
@jturkel Don't think it's a problem since it will solve itself after you update GraphiQL.
If you have a concrete scenario in mind, can you please describe it here?

Reply:
Moreover, we constantly add new features that break old GraphiQL if they are used there, e.g. repeatable directives, support for interfaces implementing other interfaces, etc.

Reply:
In the GraphiQL scenario I was thinking of (i.e. an argument disappears when marked as deprecated in older version of GraphiQL) we only control the server. The API is public so we don't control the clients (which may include things like GraphiQL or Apollo Client) so we can't easily force them to upgrade. Ideally in that scenario older clients would continue to see deprecated arguments/input field but they just wouldn't know they were deprecated. 

I could be missing something but it looks like repeatable directives and interfaces implementing other interfaces made backwards compatible changes to the introspection system (i.e. older clients wouldn't know whether a directive was repeatable or if an interface implemented another interface) but servers using those new features wouldn't break existing clients that didn't know about them.